src/js/utils/playerutils.ts

```diff
@@ -1,7 +1,12 @@
 import { version } from 'version';
 import { isFileProtocol } from 'utils/validator';
 
-export const getScriptPath = function(scriptName) {
+declare global {
+    const __SELF_HOSTED__: boolean;
+    const __REPO__: string;
```

Comment:
Could we declare all of these compiler replaced arguments in a single place? I like that they are grouped in webpack and karma configs, and would like to see them grouped together for TS as well. This is one case where I wouldn't mind a separate globals.ts(.d?) file, even if that strays from our rule of defining types in the file they are implemented in. Otherwise globals get dropped all over and it's easy to look track of them.

Reply:
yeah this is a great idea - will make a sep file